fix mispellings and add minor corrections

Mistobaan opened the pull request:
The main change of the PR is the change of interface from a single Reader(, isText) to a more (IMHO) go idiomatic `ReadFromBuffer` and `ReadFromString`

tensorflow-jenkins commented:
Can one of the admins verify this patch?

Mistobaan committed:
fix mispless and add minor corrections

alonsovidales commented:
Awesome 👍

vrv commented on tensorflow/contrib/go/README.md:
```diff
@@ -15,7 +15,7 @@ also to build, load and run Graphs.
 
 This package depends on the TensorFlow shared libraries, in order to compile
-this libraries follow the [Installing fromsources](https://www.tensorflow.org/versions/r0.8/get_started/os_setup.html#installing-from-sources)
+this libraries follow the [Installing fromsources](https://www.tensorflow.org/versions/r0.9/get_started/os_setup.html#installing-from-sources)
```
https://www.tensorflow.org/get_started/os_setup.html#installing-from-sources

use that link and it'll redirect to the latest stable version

vrv commented on tensorflow/contrib/go/graph.go:
```diff
@@ -118,16 +118,16 @@ func NewGraph() *Graph {
 // from the reader is provided in Text format.
 func NewGraphFromReader(reader io.Reader, asText bool) (*Graph, error) {
-	graphStr, err := ioutil.ReadAll(reader)
+	b, err := ioutil.ReadAll(reader)
```
'b' seems like a less useful variable name than graphStr was

Mistobaan committed:
fix documentation link

Mistobaan committed:
remove non idiomatic NewGraphFromReader(reader, bool) and add NewGraphFromBuffer and New Graph

NewGraphFromReader(reader, bool) and add NewGraphFromBuffer and NewGraphFromString.
The resulting code feels more elegant.  The user is responsible to manage the input.

vrv commented:
(ping when ready to review)

Mistobaan commented:
@vrv I think this PR is good. I tried to add some other things like the go bazel rules to build the package uzing bazel but I will wait for this PR to be merged and the branch to be rebased.

vrv commented:
@tensorflow-jenkins test this please

I have no idea if we even have tests for this or what happens, but trying it anyway.

Mistobaan commented:
@vrv there are few tests, but I don't think they are linked to the CI system.

martinwicke commented:
This didn't work, I guess.

Mistobaan commented:
@martinwicke can you guys rebase origin/go on top of origin/master ? a bunch of PR are trying to catch up the go branch to the master one.

martinwicke commented:
I rebased the go branch on top of current master.

Mistobaan commented:
🎉  Thanks ! 🎉

Mistobaan commented:
PR rebased too.

danmane commented:
@vrv you've looked at this in the past. should we merge it?

vrv commented:
I dunno, there are no tests triggered for this branch, so merging might break the branch.  @Mistobaan can fix it I guess if there are problems :)

Mistobaan commented:
👍
